src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -58,12 +67,24 @@ public static String formatChanceValue(int outputChance) {
      */
     private final boolean hidden;
 
-    private final Map<String, Object> recipeProperties;
+    private static final List<RecipeProperty> registeredProperties = new ArrayList<>();
+
+    private final List<RecipeProperty> recipeProperties;
 
     public Recipe(List<CountableIngredient> inputs, List<ItemStack> outputs, List<ChanceEntry> chancedOutputs,
```

Comment:
Right now `TemperatureProperty` is in the common package because `CoilType` is in there, for this I would have to move it to api, and I am not sure how to do that. What do you suggest?

Reply:
I don't see any reasons why `TemperatureProperty` should be in API. If this function takes `RecipeProperties` then as long `TemperatureProperty` extends it then it can be passed in.